conan/tools/meson/toolchain.py

```diff
@@ -396,6 +396,14 @@ def _sanitize_format(v):
         # These link_args have already the LDFLAGS env value so let's add only the new possible ones
         self.objc_link_args.extend(apple_flags + extra_flags["ldflags"])
         self.objcpp_link_args.extend(apple_flags + extra_flags["ldflags"])
+        # Meson properties
+        sys_root = self._conanfile.conf.get("tools.build:sysroot", check_type=str)
+        if sys_root:
+            self.properties["sys_root"] = sys_root
+            self.c_args.append("--sysroot=" + sys_root)
+            self.cpp_args.append("--sysroot=" + sys_root)
+            self.c_link_args.append("--sysroot=" + sys_root)
+            self.cpp_link_args.append("--sysroot=" + sys_root)
```

Comment:
If ``sys_root`` is a property, does it still need to be added as a flag? Wouldn't Meson take care of that automatically?

Reply:
It's needed: https://mesonbuild.com/Cross-compilation.html#properties

> Since 0.52.0 The sys_root property may point to the root of the host system path (the system that will run the compiled binaries). This is used internally by Meson to set the PKG_CONFIG_SYSROOT_DIR environment variable for pkg-config. If this is unset the host system is assumed to share a root with the build system.

The `sys_root` property only configures the `PKG_CONFIG_SYSROOT_DIR`, nothing more. 

But how Meson users manage `--sysroot` then?

They pass to `c_flags` and `cxx_flags`:

- https://github.com/mesonbuild/meson/blob/24a96a3c95ea935c169b038ed91bf6379184ad66/docs/markdown/Machine-files.md?plain=1#L67
- https://github.com/mesonbuild/meson/issues/7722

----

On the other hand, Conan users could have it defined in `tools.build:cflags` and so on.